python/lgbserver/lgbserver/model.py

```diff
@@ -51,14 +55,39 @@ def load(self) -> bool:
         self.ready = True
         return self.ready
 
-    def predict(self, payload: Dict, headers: Dict[str, str] = None) -> Dict:
+    def predict(self, payload: Union[Dict, InferRequest], headers: Dict[str, str] = None) -> Union[Dict, InferResponse]:
         try:
-            dfs = []
-            for input in payload['inputs']:
-                dfs.append(pd.DataFrame(input, columns=self._booster.feature_name()))
-            inputs = pd.concat(dfs, axis=0)
+            if isinstance(payload, Dict):
+                dfs = []
+                for input in payload['inputs']:
+                    dfs.append(pd.DataFrame(
+                        input, columns=self._booster.feature_name()))
+                inputs = pd.concat(dfs, axis=0)
+
+                result = self._booster.predict(inputs)
+                return {"predictions": result.tolist()}
```

Comment:
@yuzisun should we keep this like this? can we replace with inferrequest here to avoid hard coding key/dict?

Reply:
will address in https://github.com/kserve/kserve/pull/2748